frontend/contexts/userContext.tsx

```diff
@@ -64,7 +64,7 @@ export const UserProvider = ({ children }: { children: ReactNode }) => {
     () => ({
       user: user.data,
       privyUser: privyUser || undefined,
-      isLoading: !ready || (!!privyUser && user.isLoading),
```

Comment:
The reason I had this condition is because when privyUser is null, user.isLoading is never true. The call to getCurrentUser has a dependency on privyUser.id

If privyUser is null, the call is never made and it never goes to loading state.
And I think there is a "frame" in which ready is true, privyUser is still null, and user.isLoading is also false. So this would make a redirection even if the user was not loaded yet.

If you change the condition, make sure that it doesn't happen.